common/persistence/cassandra/matching_task_store.go

```diff
@@ -522,52 +522,46 @@ func (d *MatchingTaskStore) UpdateTaskQueueUserData(
 ) error {
 	batch := d.Session.NewBatch(gocql.UnloggedBatch).WithContext(ctx)
 
-	if request.Version == 0 {
-		batch.Query(templateInsertTaskQueueUserDataQuery,
-			request.NamespaceID,
-			request.TaskQueue,
-			request.UserData.Data,
-			request.UserData.EncodingType.String(),
-		)
-	} else {
-		batch.Query(templateUpdateTaskQueueUserDataQuery,
-			request.UserData.Data,
-			request.UserData.EncodingType.String(),
-			request.Version+1,
-			request.NamespaceID,
-			request.TaskQueue,
-			request.Version,
-		)
-	}
-	for _, buildId := range request.BuildIdsAdded {
-		batch.Query(templateInsertBuildIdTaskQueueMappingQuery, request.NamespaceID, buildId, request.TaskQueue)
-	}
-	for _, buildId := range request.BuildIdsRemoved {
-		batch.Query(templateDeleteBuildIdTaskQueueMappingQuery, request.NamespaceID, buildId, request.TaskQueue)
+	for taskQueue, update := range request.Updates {
+		if update.Version == 0 {
+			batch.Query(templateInsertTaskQueueUserDataQuery,
+				request.NamespaceID,
+				taskQueue,
+				update.UserData.Data,
+				update.UserData.EncodingType.String(),
+			)
+		} else {
+			batch.Query(templateUpdateTaskQueueUserDataQuery,
+				update.UserData.Data,
+				update.UserData.EncodingType.String(),
+				update.Version+1,
+				request.NamespaceID,
+				taskQueue,
+				update.Version,
+			)
+		}
+		for _, buildId := range update.BuildIdsAdded {
+			batch.Query(templateInsertBuildIdTaskQueueMappingQuery, request.NamespaceID, buildId, taskQueue)
+		}
+		for _, buildId := range update.BuildIdsRemoved {
+			batch.Query(templateDeleteBuildIdTaskQueueMappingQuery, request.NamespaceID, buildId, taskQueue)
+		}
 	}
 
-	previous := make(map[string]interface{})
+	previous := make(map[string]any)
 	applied, iter, err := d.Session.MapExecuteBatchCAS(batch, previous)
```

Comment:
> retries/conflicts are not handled well yet: on a version conflict, all updates in the batch will fail and none will be retried, but the non-conflicting ones should be retried

From a user perspective, is it acceptable to be released like that?

Reply:
Fixed now